charts/linkerd2/templates/linkerd-values.yaml

```diff
@@ -0,0 +1,35 @@
+{{ if or (.Values.tracing.enabled) -}}
+{{- /*
```

Comment:
The problem is that, Now Imagine the same logic being put by each and every add-on into the `linkerd-values`, It would definitely cause a lot of consistency problems and a lot of unnecessary work into `linkerd-values`. We just want to have a way to get the add-on fields into `linkerd-values`

If we just go with the `duplicated` logic, then
And the installation is through `CLI` you only get the `global` once, but if it is through helm, for each add-on you get both `partials` and `global` values and each `partial` will again have its own `global` (This is the helm way of sending values to sub-charts). It makes the whole `linkerd-values` cm a mess.

However we still have to remove the Private Keys present in `global`, thus requiring processing again.
Example gist [here](https://gist.github.com/Pothulapati/f8dde2d31318f69f0e2b58e5b1d81090) for multiple add-ons  with helm install

Reply:
My main concern is that this block of go template code is not unit testable, and debugging it is going to be a pain. Can you share the approach you have been using to debug this?

> The problem is that, Now Imagine the same logic being put by each and every add-on into the linkerd-values, 

With or without the go template code, we will still need to add a YAML block for each new addon components. I assume it's just going to be like:
```
data:
  values: |-
    tracing:
      collector:
  {{- toYaml $.Values.tracing.collector | trim | nindent 8}}
      jaeger:
  {{- toYaml $.Values.tracing.collector | trim | nindent 8}}
    monitoring:
    grafana:
  {{- toYaml $.Values.monitoring.grafana | trim | nindent 8}}
    prometheus:
  {{- toYaml $.Values.monitoring.prometheus | trim | nindent 8}}
```
Do you foresee something more complicated?

>  but if it is through helm, for each add-on you get both partials and global values and each partial will again have its own global 

I only see one `global` field when tested with `helm install` and `helm template`. See YAML ]here](https://gist.github.com/Pothulapati/f8dde2d31318f69f0e2b58e5b1d81090#gistcomment-3174836).

> It would definitely cause a lot of consistency problems and a lot of unnecessary work into linkerd-values. We just want to have a way to get the add-on fields into linkerd-values

Do you foresee anyone wanting to manually change this configmap? Or will they prefer to use the `--config` option?

> However we still have to remove the Private Keys present in global, thus requiring processing again.

You mean the public trust root?